packages/integrations/vercel/README.md

```diff
@@ -262,6 +262,61 @@ You can use Vercel middleware to intercept a request and redirect before sending
 > **Warning**
 > **Trying to rewrite?** Currently rewriting a request with middleware only works for static files.
 
+### Vercel Edge Middleware with Astro middleware
+
+The `@astrojs/vercel/serverless` adapter can automatically create the vercel edge middleware when you have an Astro middleware in you code base.
+
+This is an opt-in feature, and the `build.excludeMiddleware` option needs to be set to `true`:
```

Comment:
Discussion: I talked with Matthew a bit on this. I'm still not sure configuring this option to control an integration behaviour makes sense from an API standpoint.

I think it's more natural if they can configure as:

```js
vercel({
  edgeMiddleware: true
})
```

because `build.excludeMiddleware` can't really be used for all integrations. Maybe we can polish this up in Astro 3.0 again, where I think it's nicer if this can only be configured from integration hooks.

Reply:
So you think that when passing `edgeMiddleware` to the integration, the integration itself should update the configuration and enable `build.split`?

Reply:
Yeah, but I'm not sure if it's too late now as the same goes for `build.split`. Maybe better to keep this as-is and change it again in the next major.

In the future, we can move the new APIs to the `astro:config:setup` perhaps, similar to `addClientDirectives()`. Maybe named `setBuildMode({ excludeMiddleware: boolean })`.

I think `build.split` could be something always enabled in general though.